catroid/src/main/java/org/catrobat/catroid/utils/IdPool.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2016 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.utils;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class IdPool {
+	private static final IdPool INSTANCE = new IdPool();
+	private final Set<Integer> ids = new HashSet<>();
```

Comment:
> Just curious from the discussion of non-unique id's: Why don't you use hash values instead of consecutive Integers?

Puh, I'm not sure if we thought about that. At least I didn't :smile_cat:

@sjaindl How do you want to use hash values for that? :confused: Keep in mind that hash values changes when objects are renamed or modified.